src/main/vcp_hal/usbd_cdc_interface.c

```diff
@@ -384,11 +384,11 @@ uint32_t CDC_Send_DATA(const uint8_t *ptrBuffer, uint32_t sendLength)
 
     for (uint32_t i = 0; i < sendLength; i++)
     {
-        UserTxBuffer[UserTxBufPtrIn] = ptrBuffer[i];
-        UserTxBufPtrIn = (UserTxBufPtrIn + 1) % APP_TX_DATA_SIZE;
         while (CDC_Send_FreeBytes() == 0) {
```

Comment:
Mhhh, order here may not make real difference? If the while statement as it is before it will block anyhow till buffer has space again.

Reply:
Does making the variables `volatile` alone fix the problem? If so, it might be better to leave the control flow here the same as in the other drivers.